From 5b783ded59d1b6376189ccf9703d0614162e69a2 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 11 Jun 2009 02:58:06 +0200
Subject: [PATCH] CMake: Support installing into library directories other than 'lib'

This is useful on 64bit systems, where one can install 64-bit
libraries into /usr/lib64, by using 'cmake -DLIB_SUFFIX=64 .'
---
 cmake/modules/AddLLVM.cmake                  |    4 ++--
 cmake/modules/AddPartiallyLinkedObject.cmake |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/modules/AddLLVM.cmake b/cmake/modules/AddLLVM.cmake
index a21ed20..9b6c872 100755
--- a/cmake/modules/AddLLVM.cmake
+++ b/cmake/modules/AddLLVM.cmake
@@ -10,8 +10,8 @@ macro(add_llvm_library name)
     add_dependencies( ${name} ${LLVM_COMMON_DEPENDS} )
   endif( LLVM_COMMON_DEPENDS )
   install(TARGETS ${name}
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endmacro(add_llvm_library name)
 
 
diff --git a/cmake/modules/AddPartiallyLinkedObject.cmake b/cmake/modules/AddPartiallyLinkedObject.cmake
index d20666d..322f9f8 100755
--- a/cmake/modules/AddPartiallyLinkedObject.cmake
+++ b/cmake/modules/AddPartiallyLinkedObject.cmake
@@ -38,5 +38,5 @@ macro(add_partially_linked_object lib)
     set( llvm_lib_targets ${llvm_lib_targets} ${tnplo} PARENT_SCOPE )
   endif( )
   install(FILES ${pll}
-    DESTINATION lib)
+    DESTINATION lib${LIB_SUFFIX})
 endmacro(add_partially_linked_object lib)
-- 
1.6.3.2

From 799c42573852c08faccad5804555eeac3ebe1dd7 Mon Sep 17 00:00:00 2001
From: Ingmar Vanhassel <ingmar@exherbo.org>
Date: Thu, 11 Jun 2009 03:10:16 +0200
Subject: [PATCH] CMake: Support installing into library directories other than 'lib'

This is useful on 64bit systems, where one can install 64-bit
libraries into /usr/lib64, by using 'cmake -DLIB_SUFFIX=64.'
---
 tools/clang/CMakeLists.txt |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/clang/CMakeLists.txt b/tools/clang/CMakeLists.txt
index 45f3d0c..a2bc1ae 100644
--- a/tools/clang/CMakeLists.txt
+++ b/tools/clang/CMakeLists.txt
@@ -22,8 +22,8 @@ macro(add_clang_library name)
     set_target_properties(${name} PROPERTIES COMPILE_FLAGS ${cflag})
   endif(MSVC)
   install(TARGETS ${name}
-    LIBRARY DESTINATION lib
-    ARCHIVE DESTINATION lib)
+    LIBRARY DESTINATION lib${LIB_SUFFIX}
+    ARCHIVE DESTINATION lib${LIB_SUFFIX})
 endmacro(add_clang_library)
 
 macro(add_clang_executable name)
-- 
1.6.3.2

